src/Databases/DatabasesCommon.cpp

```diff
@@ -166,4 +168,66 @@ StoragePtr DatabaseWithOwnTablesBase::getTableUnlocked(const String & table_name
                     backQuote(database_name), backQuote(table_name));
 }
 
+/// Daisy: starts.
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const ASTPtr & query, const String & database_, const String & table_)
+{
+    return parseCreateQueryFromAST(query.get(), database_, table_);
+}
+
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const IAST * query, const String & database_, const String & table_)
+{
+    StorageInMemoryCreateQuery parsed_create_query;
+    if (query)
+    {
+        ASTPtr query_clone = query->clone();
+        auto * create = query_clone->as<ASTCreateQuery>();
+
+        if (!create)
+        {
+            WriteBufferFromOwnString query_buf;
+            formatAST(*query, query_buf, true);
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Query '{}' is not CREATE query", query_buf.str());
+        }
+
+        create->attach = false;
+        create->database = database_;
+        create->table = table_;
```

Comment:
Isn't `create->table` already exists for create ast ?

Reply:
because some `create->table` is `TABLE_WITH_UUID_NAME_PLACEHOLDER = '_'`, so I always assign a valid value to `create->table`